### ISSUE TITLE
JSONPath supports function evaluation from global scope

### ISSUE THREAD

**Issue by quantumlicht** [s3u/JSONPath#53]
For me this is actually not a bug, but a useful feature that could be more user friendly or implemented in a better way.

By reading the source code it seems that the eval function has access to the global scope when evaluating expressions in parenthesis.

For instance, if I first define a parsing function, say:

```javascript
parser = function(param){console.log(param);return '1,2,3,4,5'};
````

I can do something like that 
```javascript
JSONPath({},data, '$..book[0].category[(parser\(@\))]')
```

Where `data` is the same json as defined in the doc. Would it be possible to implement a regexp parser using this method?

I feel like regular expression is a missing feature in general and I would prefer not to use this hack.

**Comment by brettz9** [comment]
You might take a look at PR #4 and see if you can update. It should be helpful to have benchmarks too.

**Comment by brettz9** [comment]
Note that the commit 3d1a14accd626ce3d001726b63a88e832a25a2cc (in `master`) allows one to add a parser function to the sandbox for a more explicit inclusion (though it doesn't avoid an `eval` with access to global variables).

I have opened issue #60 suggesting that this become the required approach once we may move to evaluating in a true sandbox. The issue also seeks to avoid the need for `eval()` in those cases where filters are used but where arbitrary JavaScript is not needed.

Closing the issue now but feel free to comment further or contribute to issue #60 if that would meet your needs.
